### PR TITLE
Added tab group component

### PR DESCRIPTION
**Pull request by atwfk**
- [x] added `Tab` component with its tests, types and story
- [x] added `Tab` without icons variant
- [x] added `Tab` with disabled variant
- [x] added `TabGroup` component with its tests, types and story
- [x] added `tabGroupData` function that returns an array of objects that contains tabs data. 
- [x] responsive design

### CODE REVIEW

File: modules/shared/components/molecules/TabGroup/Tab.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  const tabClasses: string = classNames(styles['tab-default'], {
+    [styles['tab-checked']]: value === checkedValue,
+    [styles['tab-unchecked']]: value !== checkedValue,
+    [styles['tab-disabled']]: disabled,
+  });
+
+  return (
+    <label
+      htmlFor={id}
+      data-value={value}
```
Comment (omarawad0) [inline modules/shared/components/molecules/TabGroup/Tab.tsx:29]: What the use for this attribute?

Comment (atwfk) [reply modules/shared/components/molecules/TabGroup/Tab.tsx:29]: It's for future testing purposes when we lift up the state to the parent component ,
Check out the pr description

**Review by omarawad0 — APPROVED** [review]
LGTM

**Comment by atwfk** [comment]
It will work when we use the `useState` in the parent component 
But you are right, i need to figure out how this can work using storybook

**Pull request closed without merging**